### PR TITLE
Add installed and outdated to cask json output

### PR DESCRIPTION
**Pull request by zkokaja**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew typecheck` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?
I get one failure on an unrelated test `dev-cmd/tap-new:11` because `git` outputs a warning regarding the `init.defaultBranch` setting on new repos, and the test expected nothing in standard error.
- [X] Have you successfully run `brew man` locally and committed any changes?

-----

This commit adds  two new values to the `brew info --cask --json=v2` command to include whether the cask is outdated and the currently installed versions.

I had to modify one test script to match the new JSON output in order to pass.

### CODE REVIEW

File: Library/Homebrew/cask/cask.rb [hunk 1]
```diff
@@ -175,6 +175,8 @@ def to_h
         "url"            => url,
         "appcast"        => appcast,
         "version"        => version,
+        "installed"      => versions,
```
Comment (reitermarkus) [inline Library/Homebrew/cask/cask.rb:178]: This should be a boolean.

Comment (zkokaja) [reply Library/Homebrew/cask/cask.rb:178]: I was trying to match the keys with [formula.rb](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula.rb#L1849https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula.rb#L1849) which lists installed kegs with the `installed` key. I see why it makes sense as a boolean though. How about changing it to a boolean, and then adding `"installed_versions" => versions` to provide additional details?

**Comment by zkokaja** [comment]
Any suggestions regarding using `installed` as a boolean, and `installed_versions` to list out the installed versions?

**Comment by zkokaja** [comment]
Hi, I wanted to give a better description of this request and why it's needed. I think I should have opened an issue first to discuss before submitting the PR (sorry! first time contributing to open source). @reitermarkus, please let me know if you'd like me to open an issue first, or if I should make further changes. Thanks!

**Description:** The goal of this change is to add information on cask installed versions when using the JSON API, as well as to make it more similar to the JSON output for formulae. For example, `brew info --json=v2 jq` includes all this information:

```
      "installed": [
        {
          "version": "1.6",
          "used_options": [],
          "built_as_bottle": true,
          "poured_from_bottle": true,
          "runtime_dependencies": [
            {
              "full_name": "oniguruma",
              "version": "6.9.6"
            }
          ],
          "installed_as_dependency": false,
          "installed_on_request": true
        }
      ],
      "outdated": false,
```

Whereas if I run `brew info --json=v2 --cask rabbitmq` I get no output regarding what version is installed or whether it's outdated.


**Justification:** Adding these keys to the output makes it easier for downstream users of the API to use the `info` command to get a complete summary of the brew state without needing to coordinating calls between `info` and `outdated`. Also, making the API similar between formulae and casks is advantageous for consistency and consumption.

**Use case:** I plan to use this in my https://github.com/zkokaja/Brewlet application to notify users when casks are out of date, and show them the versions they have.

**Open question:** Should `installed` be a boolean and `installed_versions` be a list of installed versions? Or should `installed` be similar to the output for formulae?

File: Library/Homebrew/cask/cask.rb (hunk 1 above)
Comment (vitorgalvao) [reply Library/Homebrew/cask/cask.rb:178]: > How about changing it to a boolean, and then adding `"installed_versions" => versions` to provide additional details?

You can’t have more than one version of a cask installed at once—that would make no sense for most software. So just the boolean, and no further details necessary.

Comment (SeekingMeaning) [reply Library/Homebrew/cask/cask.rb:178]:
```suggestion
        "installed"      => versions.last,
```

File: Library/Homebrew/test/cask/cmd/list_spec.rb [hunk 2]
```diff
@@ -91,7 +91,7 @@
...
+        [{"token":"local-caffeine","name":[],"desc":null,"homepage":"https://brew.sh/","url":"file:///usr/local/Homebrew/Library/Homebrew/test/support/fixtures/cask/caffeine.zip","appcast":null,"version":"1.2.3","installed":["1.2.3"],"outdated":false,"sha256":"67cdb8a02803ef37fdbf7e0be205863172e41a561ca446cd84f0d7ab35a99d94","artifacts":[["Caffeine.app"]],"caveats":null,"depends_on":{},"conflicts_with":null,"container":null,"auto_updates":null},{"token":"local-transmission","name":["Transmission"],"desc":"BitTorrent client","homepage":"https://transmissionbt.com/","url":"file:///usr/local/Homebrew/Library/Homebrew/test/support/fixtures/cask/transmission-2.61.dmg","appcast":null,"version":"2.61","installed":["2.61"],"outdated":false,"sha256":"e44ffa103fbf83f55c8d0b1bea309a43b2880798dae8620b1ee8da5e1095ec68","artifacts":[["Transmission.app"]],"caveats":null,"depends_on":{},"conflicts_with":null,"container":null,"auto_updates":null}]
```
Comment (SeekingMeaning) [inline Library/Homebrew/test/cask/cmd/list_spec.rb:94]:
```suggestion
        [{"token":"local-caffeine","name":[],"desc":null,"homepage":"https://brew.sh/","url":"file:///usr/local/Homebrew/Library/Homebrew/test/support/fixtures/cask/caffeine.zip","appcast":null,"version":"1.2.3","installed":"1.2.3","outdated":false,"sha256":"67cdb8a02803ef37fdbf7e0be205863172e41a561ca446cd84f0d7ab35a99d94","artifacts":[["Caffeine.app"]],"caveats":null,"depends_on":{},"conflicts_with":null,"container":null,"auto_updates":null},{"token":"local-transmission","name":["Transmission"],"desc":"BitTorrent client","homepage":"https://transmissionbt.com/","url":"file:///usr/local/Homebrew/Library/Homebrew/test/support/fixtures/cask/transmission-2.61.dmg","appcast":null,"version":"2.61","installed":"2.61","outdated":false,"sha256":"e44ffa103fbf83f55c8d0b1bea309a43b2880798dae8620b1ee8da5e1095ec68","artifacts":[["Transmission.app"]],"caveats":null,"depends_on":{},"conflicts_with":null,"container":null,"auto_updates":null}]
```

File: Library/Homebrew/cask/cask.rb (hunk 1 above)
Comment (SeekingMeaning) [reply Library/Homebrew/cask/cask.rb:178]:
```suggestion
        "installed"      => versions.last,
```

File: Library/Homebrew/test/cask/cmd/list_spec.rb (hunk 2 above)
Comment (SeekingMeaning) [inline Library/Homebrew/test/cask/cmd/list_spec.rb:94]:
```suggestion
        [{"token":"local-caffeine","name":[],"desc":null,"homepage":"https://brew.sh/","url":"file:///usr/local/Homebrew/Library/Homebrew/test/support/fixtures/cask/caffeine.zip","appcast":null,"version":"1.2.3","installed":"1.2.3","outdated":false,"sha256":"67cdb8a02803ef37fdbf7e0be205863172e41a561ca446cd84f0d7ab35a99d94","artifacts":[["Caffeine.app"]],"caveats":null,"depends_on":{},"conflicts_with":null,"container":null,"auto_updates":null},{"token":"local-transmission","name":["Transmission"],"desc":"BitTorrent client","homepage":"https://transmissionbt.com/","url":"file:///usr/local/Homebrew/Library/Homebrew/test/support/fixtures/cask/transmission-2.61.dmg","appcast":null,"version":"2.61","installed":"2.61","outdated":false,"sha256":"e44ffa103fbf83f55c8d0b1bea309a43b2880798dae8620b1ee8da5e1095ec68","artifacts":[["Transmission.app"]],"caveats":null,"depends_on":{},"conflicts_with":null,"container":null,"auto_updates":null}]
```

**Review by SeekingMeaning — APPROVED** [review]
LGTM; if necessary, we could make changes in follow-up PRs

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @zkokaja!

**Comment by zkokaja** [comment]
Awesome, thank you!! 😊

**Pull request merged**